### DISCUSSION TITLE
TLS credentials not found although they are - "Error [IST0101] Referenced credentialName not found"

### DISCUSSION THREAD

**Discussion by orsharab** [istio/istio#53765] [General]
I'm using AKS with istio server mesh add on, version - 1.23, with external ingress gateway and sidecar injection enabled in default namespace.
When configuring an Istio Gateway with TLS, I receive the following error when running `istioctl analyze`:   
Error [IST0101] (Gateway default/hbs-system-portal-gateway) Referenced credentialName not found: "ingress-tls-csi"
However, the secret `ingress-tls-csi` exists in the `default` namespace and is correctly configured as a `kubernetes.io/tls` type:
![image](https://github.com/user-attachments/assets/321e854a-715f-4bf6-b9f8-1fa135b6308c)

Gateway yaml:
```yaml
apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  name: hbs-system-portal-gateway
  namespace: default
spec:
  selector:
    istio: aks-istio-ingressgateway-external
  servers:
    - port:
        number: 443
        name: https
        protocol: HTTPS
      tls:
        mode: SIMPLE
        credentialName: ingress-tls-csi
      hosts:
        - system-eastus.healthbot-dev.microsoft.com

Tried both version 1.22 and 1.23 what make me think i'm missing something.
Will appreciate any help with this

**Comment by keithmattix** [comment]
Did you run istioctl analyze with --systemNamespace=aks-istio-system?

**Comment by orsharab** [comment]
@keithmattix, I tried with --all-namespaces
![image](https://github.com/user-attachments/assets/95c7fb59-6aa6-47d1-8898-7fcf28f3b35a)

**Reply by keithmattix** [reply]
The istio addon on AKS runs in a non default namespace. For the most accurate output, run with --systemNamespace=aks-istio-system

**Comment by orsharab** [comment]
`istioctl analyze --systemNamespace=aks-istio-system` didn't work (see screnshot).
`istioctl analyze -i aks-istio-system` did (see screnshot).
![image](https://github.com/user-attachments/assets/9928347b-c455-4196-81e9-c5e356cace1c)

**Reply by keithmattix** [reply]
Ah I misread your original report. The secret needs to be put in the aks-istio-ingress namespace.

**Reply by howardjohn** [reply]
> Ah I misread your original report. The secret needs to be put in the aks-istio-ingress namespace.

Yep. It would probably be a good idea to make the analyzer say this
